src/test/java/uk/gov/ida/eventemitter/EventEmitterWithAMissingS3BucketIntegrationTest.java

```diff
@@ -0,0 +1,92 @@
+package uk.gov.ida.eventemitter;
+
+import cloud.localstack.LocalstackTestRunner;
+import com.amazonaws.services.s3.AmazonS3;
+import com.amazonaws.services.sqs.AmazonSQS;
+import com.amazonaws.services.sqs.model.Message;
+import com.google.inject.AbstractModule;
+import com.google.inject.Guice;
+import com.google.inject.Injector;
+import com.google.inject.Provides;
+import com.google.inject.util.Modules;
+import org.joda.time.DateTime;
+import org.joda.time.DateTimeZone;
+import org.junit.AfterClass;
+import org.junit.BeforeClass;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.io.PrintStream;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Optional;
+import java.util.UUID;
+
+import static org.assertj.core.api.AssertionsForClassTypes.assertThat;
+
+@RunWith(LocalstackTestRunner.class)
+public class EventEmitterWithAMissingS3BucketIntegrationTest {
+
+    private static final String KEY = "aesEncryptionKey";
+    private static final String SOURCE_QUEUE_NAME = "sourceQueueName";
+    private static final String BUCKET_NAME = "bucket.name";
+    private static final String KEY_NAME = "keyName";
+    private static Injector injector;
+    private static Optional<String> queueUrl;
+    private static Optional<AmazonSQS> sqs;
+    private static Optional<AmazonS3> s3;
+    private static ByteArrayOutputStream errorContent;
+    private static PrintStream printStream;
+
+    @BeforeClass
+    public static void setUp() {
```

Comment:
Useful test, but I wonder if there is an easier way to test it?

I think we are only really testing a single provider method, but we are doing it by spinning up the entire application and testing an indirect result.

Could we instead unit test the provider and assert that it returns the correct class at the end?